### PR TITLE
fix: [DEL-3719]: Improvements in watcher while watching delegate process

### PR DESCRIPTION
**Pull request by gaurav-jain1**
* fix: [DEL-3719]: This changeset contains following improvements:
1) Send primaryVersion while starting Delegate process (in case of on-prem)
2) Verbose logging if we get exception in watchDelegate()
3) Look for existence of delegate process before attempting to kill it.

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

[DEL-3719]: https://harness.atlassian.net/browse/DEL-3719?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32510)

### CODE REVIEW

File: 260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java [hunk 1]
```diff
@@ -922,7 +922,9 @@ private void trySocketReconnect() {
     try {
       FibonacciBackOff.executeForEver(() -> {
         RequestBuilder requestBuilder = prepareRequestBuilder();
-        return socket.open(requestBuilder.build());
+        Socket skt = socket.open(requestBuilder.build());
+        log.info("Socket status: {}", socket.status().toString());
```
Comment (arpit3107) [inline 260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java:926]: shouldnt we log skt.status() in place of socket.status()

**Review by gaurav-jain1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32510)* status: 3 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java` line 926 at r1](https://reviewable.io/reviews/harness/harness-core/32510#-N0ZYnBdUQuC2ATyE0TY-r1-926:-N0ZerRF9n4nd1jATZa7:b-7ttzpo) ([raw file](https://github.com/harness/harness-core/blob/ec838791b6a77c8d95686149e37359f290371b33/260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java#L926)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

shouldnt we log skt.status() in place of socket.status()
</blockquote></details>

Both are same, "socket.open" internally returns "this"

**Review by arpit3107 — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32510)* status: :shipit: complete! ✓

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32510)* status: :shipit: complete! ✓

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32510)* status: :shipit: complete! ✓

**Pull request merged**